### ISSUE TITLE
istanbul ignore hint misplaced when semi-colons are omitted after literal object

### ISSUE THREAD

**Issue by coreyfarrell** [istanbuljs/istanbuljs#467]
```js
const obj = {a: 'a'}

/* istanbul ignore if */
if ('b' in obj) {
  console.log('this should not be instrumented')
}
```

The output of `nyc instrument --compact=false` for this file is:
```js
/* ...clipped... */
const obj = (cov_2oqccc38jn.s[0]++, {
  a: 'a'
  /* istanbul ignore if */

});
cov_2oqccc38jn.s[1]++;

if ('b' in obj) {
  cov_2oqccc38jn.b[0][0]++;
  cov_2oqccc38jn.s[2]++;
  console.log('this should not be instrumented');
} else {
  cov_2oqccc38jn.b[0][1]++;
}
```

Somehow the hint is getting moved up.  I'm only able to reproduce this issue when the statement before the hint is a literal object assignment with at least one key and no semi-colons are used.  Not sure yet if this is a bug in istanbul-lib-instrument or if babel is doing something unexpected.

**Comment by coreyfarrell** [comment]
Appears to be the result of a regression in `@babel/parser`, will be fixed by babel/babel#10445.

**Comment by coreyfarrell** [comment]
Fixed in `@babel/parser@7.6.2`
